message-processor/src/main/java/com/bosch/iot/things/example/message/processor/transport/AmqpClient.java

```diff
@@ -0,0 +1,126 @@
+package com.bosch.iot.things.example.message.processor.transport;
+
+import io.vertx.core.Vertx;
+import io.vertx.proton.ProtonClient;
+import io.vertx.proton.ProtonClientOptions;
+import io.vertx.proton.ProtonConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.annotation.PostConstruct;
+import java.util.Objects;
+
+import static java.lang.Thread.sleep;
+
+public class AmqpClient {
+
+    private static final Logger log = LoggerFactory.getLogger(AmqpClient.class);
+
+    @Autowired
+    private Vertx vertx;
+
+    @Value(value = "${hub.client.host}")
+    private String hubHost;
+
+    @Value(value = "${hub.client.port}")
+    private Integer hubPort;
+
+    @Value(value = "${hub.client.username}")
+    private String hubUsername;
+
+    @Value(value = "${hub.client.password}")
+    private String hubPassword;
```

Comment:
Is it possible to "forward" the credentials? Then we would not have to configure it for every instance.

Reply:
I will have a look and see what I could do about it!